website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,280 @@
+    'return them to private later, but search engines (including Google’s cache) or others may access files before you do.',
+
+    selectNodes: 'Adjust your privacy settings by checking the boxes below. ' +
+    '<br><br><b>Checked</b> projects and components will be <b>public</b>.  <br><b>Unchecked</b> components will be <b>private</b>.',
+    confirmWarning: {
+        nodesPublic: 'The following projects and components will be made <b>public</b>.',
+        nodesPrivate: 'The following projects and components will be made <b>private</b>.',
+        nodesNotChangedWarning: 'No nodes were changed.'
+    }
+};
+
+/**
+ * take treebeard tree structure of nodes and get a dictionary of parent node and all its
+ * children
+ */
+function getNodesOriginal(nodeTree, nodesOriginal) {
+    var i;
+    var nodeId = nodeTree.node.id;
+    nodesOriginal[nodeId] = {
+        public: nodeTree.node.is_public,
+        id: nodeTree.node.id,
+        title: nodeTree.node.title,
+        canWrite: nodeTree.node.can_write,
+        changed: false
+    };
+
+    if (nodeTree.children) {
+        for (i in nodeTree.children) {
+            nodesOriginal = getNodesOriginal(nodeTree.children[i], nodesOriginal);
+        }
+    }
+    return nodesOriginal;
+}
+
+/**
+ * patches all the nodes in a changed state
+ * uses API v2 bulk requests
+ */
+function patchNodesPrivacy(nodes) {
+    var nodesV2Url = window.contextVars.apiV2Prefix + 'nodes/';
+    var nodesPatch = nodes.splice(0,10).map(function (node) {
+        return {
+            'type': 'nodes',
+            'id': node.id,
+            'attributes': {
+                'public': node.public
+            }
+        };
+    });
+    return $.ajax({
+        url: nodesV2Url,
+        type: 'PATCH',
+        dataType: 'json',
+        contentType: 'application/vnd.api+json; ext=bulk',
+        crossOrigin: true,
+        xhrFields: {withCredentials: true},
+        processData: false,
+        data: JSON.stringify({
+            data: nodesPatch
+        })
+    }).done(function (response) {
+        if (nodes.length === 0) {
+            return;
+        }
+        return patchNodesPrivacy(nodes);
+    }).fail(function(xhr, status, error) {
+        $osf.growl('Error', 'Unable to update project privacy');
+        Raven.captureMessage('Could not PATCH project settings.', {
+            url: nodesV2Url, status: status, error: error
+        });
+    });
+}
+
+/**
+ * view model which corresponds to nodes_privacy.mako (#nodesPrivacy)
+ *
+ * @type {NodesPrivacyViewModel}
+ */
+var NodesPrivacyViewModel = function(parentIsPublic) {
+    var self = this;
+    self.WARNING = 'warning';
+    self.SELECT = 'select';
+    self.CONFIRM = 'confirm';
+
+    var treebeardUrl = window.contextVars.node.urls.api  + 'tree/';
+    self.nodesOriginal = {};
+    self.nodesChanged = ko.observable();
+    //state of current nodes
+    self.nodesState = ko.observableArray();
```

Comment:
I can't seem to find my previous comment on this, but I don't think this is supposed to be an `observableArray`.